File: angular/MineBuildSell/src/app/shared/model/fleet-model.ts
import {CharacterModel} from './character.model';
import {TypeModel} from './type-model';
import {RoleModel} from './role.model';

export class FleetModel {

  constructor(public id: number,
              public name: string,
              public description: string,
              public locationId: number,
              public owner: number,
              public type: TypeModel,
              public typeId: number,
              public start: string,
              public duration: number,
              public restricted: boolean,
              public invitations: CharacterModel[],
              public roles: RoleModel[]) {
  }

}

File: domain/src/main/java/com/ractoc/eve/domain/fleetmanager/FleetModel.java
package com.ractoc.eve.domain.fleetmanager;

import io.swagger.annotations.ApiModel;
import lombok.Getter;
import lombok.NonNull;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.validator.constraints.Length;

@Getter
@Setter
@ToString
@ApiModel(value = "Fleet Model", description = "Contains the Fleet model information")
public class FleetModel {
    private Integer id;
    @NonNull
    private String name;
    @Length(max = 1000, message = "Description can not exceed a 1000 characters")
    private String description;

    private Integer owner;
    @NonNull
    private String start;
    private Integer duration;

    private boolean restricted = false;

    @NonNull
    private TypeModel type;

}
